Rebus.AdoNet/AdoNetTimeoutStorage.cs

```diff
@@ -105,20 +148,33 @@ public DueTimeoutsResult GetDueTimeouts()
 				{
 					while (reader.Read())
 					{
-						var sqlTimeout = DueAdoNetTimeout.Create(MarkAsProcessed, timeoutsTableName, reader);
+						var sqlTimeout = DueAdoNetTimeout.Create(MarkAsProcessed, timeoutsTableName, reader, connection, transaction);
 
 						dueTimeouts.Add(sqlTimeout);
-					}
+					}
 				}
 			}
 
-			return new DueTimeoutsResult(dueTimeouts);
+			return new DueTimeoutsResult(dueTimeouts, () => CommitAndClose(connection, transaction));
+		}
+
+		void CommitAndClose(IDbConnection connection, IDbTransaction transaction)
+		{
+			try
+			{
+				transaction.Commit();
```

Comment:
every disposal action should be performed within it's own try/catch.
Otherwise, a (failed) call to Commit, will leak an open transaction & connection..